.github/workflows/branch-release.yaml

```diff
@@ -1,57 +1,45 @@
 name: Branch Release
 
-on: push
+on:
+  push
 
 jobs:
-  branch-release:
+  branch_publish:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
 
-      - name: Set up Node.js
+      - name: Setup Node.js
         uses: actions/setup-node@v4
         with:
-          node-version: "20"
-          registry-url: https://registry.npmjs.org/
+          node-version: '20'
 
       - name: Install pnpm
         uses: pnpm/action-setup@v2
         with:
           version: latest
 
-      - name: Cache pnpm modules
-        uses: actions/cache@v4
-        with:
-          path: |
-            ~/.pnpm-store
-            node_modules
-            src/playground/node_modules
-          key: ${{ runner.os }}-pnpm-${{ hashFiles('**/pnpm-lock.yaml') }}
-          restore-keys: |
-            ${{ runner.os }}-pnpm-
-
-      - name: Install Full Project
-        run: pnpm install --frozen-lockfile
-
-      - name: Build Full Project
+      - name: Install Dependencies
+        run: pnpm install --froze
+
+      - name: Build
         run: pnpm run build
 
       - name: Set Version from Branch
         run: |
           BRANCH_NAME=${GITHUB_REF#refs/heads/}
-          BRANCH_NAME=${BRANCH_NAME//\//-} # Replace slashes with dashes
           SHORT_SHA=$(git rev-parse --short HEAD)
-          # TODO: make this dynamic https://github.com/BitteProtocol/make-agent/issues/56
-          BASE_VERSION="0.2.1" # Define a base version
+          BASE_VERSION=$(node -p "require('./package.json').version")
```

Comment:
This is no good.

Please consider following the linked issue: https://github.com/BitteProtocol/make-agent/issues/56

Also save this for another PR - it is an unrelated change.